share/sysbuild/cmake/modules/sysbuild_kconfig.cmake

```diff
@@ -24,22 +24,25 @@ else()
   # SYSBuild dedicated configuration file.
 endif()
 
-if(DEFINED SB_CONF_FILE AND NOT IS_ABSOLUTE SB_CONF_FILE)
-  cmake_path(ABSOLUTE_PATH SB_CONF_FILE BASE_DIRECTORY ${APP_DIR})
-endif()
-
-if(DEFINED SB_OVERLAY_CONFIG AND NOT IS_ABSOLUTE SB_OVERLAY_CONFIG)
-  cmake_path(ABSOLUTE_PATH SB_OVERLAY_CONFIG BASE_DIRECTORY ${APP_DIR})
-endif()
-
-if(DEFINED SB_EXTRA_CONF_FILE AND NOT IS_ABSOLUTE SB_EXTRA_CONF_FILE)
-  cmake_path(ABSOLUTE_PATH SB_EXTRA_CONF_FILE BASE_DIRECTORY ${APP_DIR})
-endif()
-
 if(NOT DEFINED SB_EXTRA_CONF_FILE AND DEFINED SB_OVERLAY_CONFIG)
   set(SB_EXTRA_CONF_FILE ${SB_OVERLAY_CONFIG})
 endif()
 
+# Let SB_CONF_FILE and SB_EXTRA_CONF_FILE be relative to APP_DIR.
+# Either variable can be a list of paths, so we must make all of them absolute.
+foreach(conf_file_var SB_CONF_FILE SB_EXTRA_CONF_FILE)
+  if(DEFINED ${conf_file_var})
+    string(CONFIGURE "${${conf_file_var}}" conf_file_expanded)
+    set(${conf_file_var} "")
```

Comment:
Nothing wrong, so feel free to not follow the comment, but the two vars `SB_CONF_FILE` and `SB_EXTRA_CONF_FILE` are only used in this file as input to Zephyr Kconfig CMake module here:
https://github.com/zephyrproject-rtos/zephyr/blob/fdaba20b2cef812d611d2d2f665d43c9f20c2833/share/sysbuild/cmake/modules/sysbuild_kconfig.cmake#L55
https://github.com/zephyrproject-rtos/zephyr/blob/fdaba20b2cef812d611d2d2f665d43c9f20c2833/share/sysbuild/cmake/modules/sysbuild_kconfig.cmake#L62

so after path conversion then you could decide to populate a new and clean variable instead of making the cache variable into a local scope variable.

Reply:
I suppose you mean I could set `CONF_FILE` in this loop immediately, instead of updating `SB_CONF_FILE` first, but I think setting `CONF_FILE` at the bottom is cleaner.

I also thought for a moment that I could drop the `set("")` line this way, but `CONF_FILE` is not guaranteed to be "new and clean" at this point.

Reply:
nope, I actually meant a new local and clean variable, like this:
```
foreach(conf_file_var SB_CONF_FILE SB_EXTRA_CONF_FILE)
  if(DEFINED ${conf_file_var})
    string(CONFIGURE "${${conf_file_var}}" conf_file_expanded)
    foreach(conf_file ${conf_file_expanded})
      cmake_path(ABSOLUTE_PATH conf_file BASE_DIRECTORY ${APP_DIR})
      list(APPEND local_${conf_file_var} ${conf_file})
    endforeach()
  endif()
endforeach()

...
if(NOT DEFINED SB_CONF_FILE)
  # If there is no SB_CONF_FILE, then use empty.conf to make kconfiglib happy.
  # Not adding it to CMake cache ensures that a later created sysbuild.conf
  # will be automatically detected.
  set(local_SB_CONF_FILE ${CMAKE_CURRENT_BINARY_DIR}/empty.conf)
endif()
...
set(CONF_FILE ${local_SB_CONF_FILE})
set(EXTRA_CONF_FILE "${local_SB_EXTRA_CONF_FILE}")
```

that way the code is not hiding the cached `SB_CONF_FILE` and friend, and in traces you have the possibility of following also the `local_SB_CONF_FILE` variable.